contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java

```diff
@@ -16,39 +16,56 @@
 
 package io.opentelemetry.contrib.jmxmetrics;
 
-import io.opentelemetry.api.OpenTelemetry;
-import io.opentelemetry.api.common.Labels;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.metrics.AsynchronousInstrument;
 import io.opentelemetry.api.metrics.DoubleCounter;
 import io.opentelemetry.api.metrics.DoubleSumObserver;
 import io.opentelemetry.api.metrics.DoubleUpDownCounter;
 import io.opentelemetry.api.metrics.DoubleUpDownSumObserver;
 import io.opentelemetry.api.metrics.DoubleValueObserver;
 import io.opentelemetry.api.metrics.DoubleValueRecorder;
+import io.opentelemetry.api.metrics.GlobalMetricsProvider;
 import io.opentelemetry.api.metrics.LongCounter;
 import io.opentelemetry.api.metrics.LongSumObserver;
 import io.opentelemetry.api.metrics.LongUpDownCounter;
 import io.opentelemetry.api.metrics.LongUpDownSumObserver;
 import io.opentelemetry.api.metrics.LongValueObserver;
 import io.opentelemetry.api.metrics.LongValueRecorder;
 import io.opentelemetry.api.metrics.Meter;
-import io.opentelemetry.exporter.logging.LoggingMetricExporter;
-import io.opentelemetry.exporter.otlp.OtlpGrpcMetricExporter;
-import io.opentelemetry.exporter.prometheus.PrometheusCollector;
-import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.api.metrics.common.LabelsBuilder;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.common.InstrumentDescriptor;
+import io.opentelemetry.sdk.metrics.common.InstrumentType;
+import io.opentelemetry.sdk.metrics.common.InstrumentValueType;
 import io.opentelemetry.sdk.metrics.data.MetricData;
 import io.opentelemetry.sdk.metrics.export.MetricExporter;
-import io.opentelemetry.sdk.metrics.export.MetricProducer;
-import io.opentelemetry.sdk.testing.exporter.InMemoryMetricExporter;
-import io.prometheus.client.exporter.HTTPServer;
-import java.io.IOException;
+import io.opentelemetry.sdk.metrics.testing.InMemoryMetricExporter;
 import java.util.Collection;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
 
 public class GroovyMetricEnvironment {
 
+  private final SdkMeterProvider meterProvider;
   private final Meter meter;
+
+  // will only be `inmemory` since otel-java autoconfigure sdk extension manages other exporters
   private MetricExporter exporter;
-  private HTTPServer prometheusServer;
+
+  // Observer updaters can only be specified in the builder as of v0.13.0, so to work with our model
+  // of running groovy scripts on an interval a reference to the desired updater should be held and
+  // updated w/ each instrument creation call.  Otherwise no observed changes in MBean availability
+  // would be possible.  These registry stores are maps of instrument descriptor hashes to updater
+  // consumer references.
+  private final ConcurrentHashMap<
```

Comment:
For these two, could you make the member type just a `Map`?

Reply:
Updated